python/tvm/build_module.py

```diff
@@ -168,6 +236,17 @@ def get_binds(args, binds=None):
     return binds, arg_list
 
 
+def dump_ir(func):
```

Comment:
this decorator function is confusing. Simply add a DumpIR object to a member of BuildConfig, and call it during exit enter.

Reply:
do not use decorator function unless we need it in a public API

Reply:
How to call DumpIR in BuildConfig, use with style?

Reply:
create a DumpIR object in constructor, call its enter and exit explicitly in BuildConfig's enter and exit